src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs

```diff
@@ -193,6 +200,92 @@ private List<string> ResolveByExternalReferencePackages(List<string> uidList, Co
             return list;
         }
 
+        private async Task<List<string>> ResolveByXRefServiceAsync(List<string> uidList, ConcurrentDictionary<string, XRefSpec> externalXRefSpec)
+        {
+            if (XRefServiceUrls == null || XRefServiceUrls.Length == 0)
+            {
+                Logger.LogWarning($"You haven't provide an xrefservice item in docfx.json or command options!");
+                return uidList;
+            }
+            string requestUrl = XRefServiceUrls[0];
+            var list = new List<string>();
+            int pieceSize = 1000;
+            using (var client = new HttpClient())
+            {   
+                try
+                {
+                    client.BaseAddress = new Uri(requestUrl);
+                }
+                catch (UriFormatException e)
+                {
+                    Logger.LogWarning($"Ignore invalid url: {requestUrl}." + e.Message);
+                    return uidList;
+                }
+                catch (ArgumentException e)
+                {
+                    Logger.LogWarning($"Ignore invalid url: {requestUrl}." + e.Message);
+                    return uidList;
+                }
```

Comment:
An ArgumentException will be thrown if the url is invalid which not starts with http or https scheme, or instead use if statement but is there some other condition happen?

---
In reply to: [126846286](https://github.com/dotnet/docfx/pull/1837#discussion_r126846286) [](ancestors = 126846286)